src/device.rs

```diff
@@ -48,6 +48,26 @@ pub unsafe extern "C" fn wgpuAdapterRequestDevice(
     (callback.unwrap())(device_id, userdata);
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn wgpuAdapterGetProperties(
+    adapter: id::AdapterId,
+    properties: &mut native::WGPUAdapterProperties,
+) {
+    //let maybe_props: Result<wgt::AdapterInfo, wgc::instance::InvalidAdapter> = GLOBAL.adapter_get_info(adapter);
+
+    let maybe_props = gfx_select!(adapter => GLOBAL.adapter_get_info(adapter));
+    match maybe_props {
+        Ok(props) => {
+            //properties.name = std::ffi::CString::new(props.name).unwrap().as_ptr();
```

Comment:
Are we supposed to create the memory and pass the pointer, or should the user prepare some memory for us to write to?

Reply:
Either way it might be smart to leave a comment in the header or add an example for this so that is clear who owns what

Reply:
I think the memory should be created by wgpu-native, because there it is know how large the string is.